l2-contracts/contracts/L1Messenger.sol

```diff
@@ -47,8 +47,10 @@ contract L1Messenger is IL1Messenger {
     function burnGas(bytes calldata _message) internal {
         uint256 gasToBurn = estimateL1MessageGas(_message.length);
 
-        uint256 endGas = gasleft() - gasToBurn;
-        require(endGas > 0, NotEnoughGasSupplied());
+        // If not enough gas to burn the desired amount, revert with your custom error
+        if (gasleft() <= gasToBurn) {
```

Comment:
Do we want to reflect 63/64 rule?


```solidity
if (gasleft() * 63 / 64 < gasToBurn) {
```

It won't guarantee the exact amount since we'll spend some gas on the call itself, but this is ok imo.